sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/specialized/BlobInputStream.java

```diff
@@ -89,4 +102,26 @@ protected synchronized ByteBuffer dispatchRead(final int readLength, final long
         }
     }
 
+    private static BlobProperties buildBlobProperties(BlobDownloadHeaders hd) {
+        if (hd == null) {
+            return null;
+        }
+        return new BlobProperties(null, hd.getLastModified(), hd.getETag(),
+            hd.getContentLength() == null ? 0 : hd.getContentLength(), hd.getContentType(), null,
+            hd.getContentEncoding(), hd.getContentDisposition(), hd.getContentLanguage(), hd.getCacheControl(),
+            hd.getBlobSequenceNumber(), hd.getBlobType(), hd.getLeaseStatus(), hd.getLeaseState(),
+            hd.getLeaseDuration(), hd.getCopyId(), hd.getCopyStatus(), hd.getCopySource(), hd.getCopyProgress(),
+            hd.getCopyCompletionTime(), hd.getCopyStatusDescription(), hd.isServerEncrypted(),
+            null, null, null, null, null, hd.getEncryptionKeySha256(), null, hd.getMetadata(),
+            hd.getBlobCommittedBlockCount());
+    }
+
+    /**
+     *
```

Comment:
Extra space.

Looking at this, I wonder if it actually would be good to do a getProperties call if they try to access the properties before reading any data instead of returning null. What do you think?

Reply:
Good idea, I'll change it to do that and document it